diff --git a/crypto/frontend/pages/settings.js b/crypto/frontend/pages/settings.js
new file mode 100644
index 0000000000000000000000000000000000000000..09b0180341866eebce62d54a8c9e5f309e7e2426
--- /dev/null
+++ b/crypto/frontend/pages/settings.js
@@ -0,0 +1 @@
+export {Settings as default} from '@/scenes';
diff --git a/crypto/frontend/scenes/dashboard/DashboardTickerItem.js b/crypto/frontend/scenes/dashboard/DashboardTickerItem.js
new file mode 100644
index 0000000000000000000000000000000000000000..d826184545a625bbc196a4bec406e6740ad67967
--- /dev/null
+++ b/crypto/frontend/scenes/dashboard/DashboardTickerItem.js
@@ -0,0 +1,79 @@
+import {Link, Stack, Typography} from '@mui/material';
+import NextLink from 'next/link';
+import {memo} from 'react';
+import {graphql, useFragment} from 'react-relay';
+
+import {
+  direction,
+  findClosestAccessibleColor,
+  formatCurrency,
+  formatPercent,
+} from '@/utils';
+
+const Symbol = memo(function Symbol({value, color}) {
+  return (
+    <Typography
+      variant="caption"
+      sx={(theme) => ({
+        color: findClosestAccessibleColor(
+          color,
+          theme.palette.background.default,
+          3,
+        ),
+        fontSize: theme.typography.fontSizeTiny,
+        fontWeight: theme.typography.fontWeightMedium,
+        lineHeight: 1,
+      })}
+    >
+      {value}
+    </Typography>
+  );
+});
+
+const Price = ({value, options, locales}) => (
+  <Typography
+    variant="caption"
+    sx={(theme) => ({
+      color: theme.palette.text.primary,
+      fontSize: theme.typography.fontSizeMedium,
+      fontWeight: theme.typography.fontWeightMedium,
+      lineHeight: 1,
+    })}
+  >
+    {formatCurrency(value, options, locales)}
+  </Typography>
+);
+
+const Change = ({value, options, locales}) => (
+  <Typography
+    variant="caption"
+    sx={(theme) => ({
+      color: theme.palette.trend[direction(value)],
+      fontSize: theme.typography.fontSizeTiny,
+      fontWeight: theme.typography.fontWeightMedium,
+      lineHeight: 1,
+    })}
+  >
+    {formatPercent(value, options, locales)}
+  </Typography>
+);
+
+export default memo(function DashboardTickerItem({fragmentRef}) {
+  // ... ?
+  //
+  // return (
+  //   <NextLink
+  //     href="/currencies/[symbol]"
+  //     as={`/currencies/${asset.symbol}`}
+  //     passHref
+  //   >
+  //     <Link underline="none">
+  //       <Stack direction="column" spacing={1} minWidth={100}>
+  //         <Symbol value={asset.symbol} color={asset.color} />
+  //         <Price value={price.lastPrice} options={{currency: price.currency}} />
+  //         <Change value={price.change24Hour} />
+  //       </Stack>
+  //     </Link>
+  //   </NextLink>
+  // );
+});
diff --git a/crypto/frontend/scenes/settings/index.js b/crypto/frontend/scenes/settings/index.js
new file mode 100644
index 0000000000000000000000000000000000000000..decad83e3475289c4703057a581da1dbc0656752
--- /dev/null
+++ b/crypto/frontend/scenes/settings/index.js
@@ -0,0 +1,15 @@
+import {Suspense} from 'react';
+
+import {ActivityIndicator, ErrorBoundaryWithRetry} from '@/components';
+
+import SettingsContainer from './SettingsContainer';
+
+export const Settings = ({symbol}) => (
+  <ErrorBoundaryWithRetry key={symbol}>
+    {({cacheBuster}) => (
+      <Suspense fallback={<ActivityIndicator />}>
+        <SettingsContainer cacheBuster={cacheBuster} />
+      </Suspense>
+    )}
+  </ErrorBoundaryWithRetry>
+);
diff --git a/crypto/frontend/scenes/settings/SettingsLinks.js b/crypto/frontend/scenes/settings/SettingsLinks.js
new file mode 100644
index 0000000000000000000000000000000000000000..b7a567f5fdffae127d3348802e9ddf8530572e2e
--- /dev/null
+++ b/crypto/frontend/scenes/settings/SettingsLinks.js
@@ -0,0 +1,75 @@
+import {
+  AccordionActions,
+  AccordionDetails,
+  Button,
+  Stack,
+  TextField,
+} from '@mui/material';
+import {memo, useRef} from 'react';
+
+import {AccordionGroup} from '@/components';
+import {Config, syncWithStorage} from '@/config';
+import {LinksIcon} from '@/icons';
+
+export default memo(function SettingsLinks({active, onChange}) {
+  const httpRef = useRef(null);
+  const wsRef = useRef(null);
+
+  return (
+    <AccordionGroup
+      id="links"
+      icon={LinksIcon}
+      title="Links"
+      active={active}
+      onChange={onChange}
+    >
+      <form
+        name="links"
+        onSubmit={(e) => {
+          e.preventDefault();
+
+          const http = httpRef.current?.value;
+          const ws = wsRef.current?.value;
+
+          syncWithStorage({
+            HTTP_ENDPOINT: http,
+            WS_ENDPOINT: ws,
+          });
+        }}
+      >
+        <AccordionDetails id="panel-links-content" sx={{p: 6}}>
+          <Stack direction="column" gap={8}>
+            <TextField
+              inputRef={httpRef}
+              label="HTTP Endpoint"
+              defaultValue={Config.HTTP_ENDPOINT}
+              autoComplete="http"
+              helperText="The URL of the server to send GraphQL queries over HTTP."
+              required
+              fullWidth
+            />
+            <TextField
+              inputRef={wsRef}
+              label="WS Endpoint"
+              defaultValue={Config.WS_ENDPOINT}
+              autoComplete="ws"
+              helperText="The URL of the server to send GraphQL queries over WebSocket."
+              required
+              fullWidth
+            />
+          </Stack>
+        </AccordionDetails>
+        <AccordionActions
+          sx={{flexDirection: 'row-reverse', justifyContent: 'flex-start'}}
+        >
+          <Button type="submit" color="primary">
+            Save
+          </Button>
+          <Button type="reset" color="secondary">
+            Cancel
+          </Button>
+        </AccordionActions>
+      </form>
+    </AccordionGroup>
+  );
+});
diff --git a/crypto/frontend/scenes/index.js b/crypto/frontend/scenes/index.js
new file mode 100644
index 0000000000000000000000000000000000000000..1bf359a678d71f0f3c9d18a81f4f9a146ae64db2
--- /dev/null
+++ b/crypto/frontend/scenes/index.js
@@ -0,0 +1,2 @@
+export * from './dashboard';
+export * from './settings';
diff --git a/crypto/frontend/scenes/dashboard/index.js b/crypto/frontend/scenes/dashboard/index.js
new file mode 100644
index 0000000000000000000000000000000000000000..312b9c8e79235017f7e38340d94ce7d54a77e940
--- /dev/null
+++ b/crypto/frontend/scenes/dashboard/index.js
@@ -0,0 +1,15 @@
+import {Suspense} from 'react';
+
+import {ActivityIndicator, ErrorBoundaryWithRetry} from '@/components';
+
+import DashboardContainer from './DashboardContainer';
+
+export const Dashboard = () => (
+  <ErrorBoundaryWithRetry>
+    {({cacheBuster}) => (
+      <Suspense fallback={<ActivityIndicator />}>
+        <DashboardContainer cacheBuster={cacheBuster} />
+      </Suspense>
+    )}
+  </ErrorBoundaryWithRetry>
+);
diff --git a/crypto/frontend/scenes/settings/SettingsCredentials.js b/crypto/frontend/scenes/settings/SettingsCredentials.js
new file mode 100644
index 0000000000000000000000000000000000000000..d63292876326cce268eee44406f140edf9e0b9c2
--- /dev/null
+++ b/crypto/frontend/scenes/settings/SettingsCredentials.js
@@ -0,0 +1,76 @@
+import {
+  AccordionActions,
+  AccordionDetails,
+  Button,
+  Stack,
+  TextField,
+} from '@mui/material';
+import {memo, useRef} from 'react';
+
+import {AccordionGroup, PasswordField} from '@/components';
+import {Config, syncWithStorage} from '@/config';
+import {CredentialsIcon} from '@/icons';
+
+export default memo(function SettingsCredentials({active, onChange}) {
+  const usernameRef = useRef(null);
+  const passwordRef = useRef(null);
+
+  return (
+    <AccordionGroup
+      id="credentials"
+      icon={CredentialsIcon}
+      title="Credentials"
+      active={active}
+      onChange={onChange}
+    >
+      <form
+        name="credentials"
+        onSubmit={(e) => {
+          e.preventDefault();
+
+          const username = usernameRef.current?.value?.trim();
+          const password = passwordRef.current?.value?.trim();
+          const token =
+            username && password
+              ? Buffer.from(`${username}:${password}`).toString('base64')
+              : null;
+
+          syncWithStorage({
+            USERNAME: username,
+            PASSWORD: password,
+            AUTH_TOKEN: token,
+          });
+        }}
+      >
+        <AccordionDetails id="panel-credentials-content" sx={{p: 6}}>
+          <Stack direction="column" gap={8}>
+            <TextField
+              inputRef={usernameRef}
+              label="Username"
+              defaultValue={Config.USERNAME}
+              autoComplete="username"
+              fullWidth
+            />
+            <PasswordField
+              inputRef={passwordRef}
+              label="Password"
+              defaultValue={Config.PASSWORD}
+              autoComplete="current-password"
+              fullWidth
+            />
+          </Stack>
+        </AccordionDetails>
+        <AccordionActions
+          sx={{flexDirection: 'row-reverse', justifyContent: 'flex-start'}}
+        >
+          <Button type="submit" color="primary">
+            Save
+          </Button>
+          <Button type="reset" color="secondary">
+            Cancel
+          </Button>
+        </AccordionActions>
+      </form>
+    </AccordionGroup>
+  );
+});
diff --git a/crypto/frontend/pages/index.js b/crypto/frontend/pages/index.js
index 8b3e2fa25f805f5e59f6da3bc19b8f1365bbb124..f5b3bfcc5a809b3d23c30b8a2d16fc34f5eabf4a 100644
--- a/crypto/frontend/pages/index.js
+++ b/crypto/frontend/pages/index.js
@@ -1 +1 @@
-export {Dummy as default} from '@/components';
+export {Dashboard as default} from '@/scenes';
diff --git a/crypto/frontend/client/index.js b/crypto/frontend/client/index.js
new file mode 100644
index 0000000000000000000000000000000000000000..ba7f1d72553d0d6282bda51657fa7260a2aa6322
--- /dev/null
+++ b/crypto/frontend/client/index.js
@@ -0,0 +1,286 @@
+import extractFiles from 'extract-files/extractFiles.mjs';
+import isExtractableFile from 'extract-files/isExtractableFile.mjs';
+import {createClient} from 'graphql-ws';
+import {meros} from 'meros/browser';
+import {useMemo} from 'react';
+import {
+  Environment,
+  Network,
+  Observable,
+  RecordSource,
+  Store,
+} from 'relay-runtime';
+
+import {Config} from '@/config';
+import {isAsyncIterable, merge, pause} from '@/utils';
+
+let relayEnvironment;
+
+const ErrorMessages = {
+  FAILED_FETCH: 'Failed to fetch',
+  ERROR_FETCH: 'Error in fetch',
+  UNWORKABLE_FETCH: 'Unworkable fetch',
+  SOCKET_CLOSED: 'Socket closed',
+  GRAPHQL_ERRORS: 'GraphQL error',
+};
+
+class NetworkError extends Error {
+  constructor(message, options) {
+    super(message, options);
+
+    this.name = 'NetworkError';
+
+    if (options) {
+      const {cause, ...meta} = options;
+
+      Object.assign(this, meta);
+    }
+  }
+}
+
+const fetchFn = (operation, variables, cacheConfig, uploadables) => {
+  const httpEndpoint = Config.HTTP_ENDPOINT;
+  const authToken = Config.AUTH_TOKEN;
+
+  return Observable.create((sink) => {
+    const init = {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        Authorization: authToken ? `basic ${authToken}` : undefined,
+      },
+    };
+
+    const {clone, files} = extractFiles(
+      {
+        id: operation.id ?? undefined,
+        query: operation.text ?? undefined,
+        variables,
+      },
+      isExtractableFile,
+    );
+
+    if (files.size) {
+      const form = new FormData();
+
+      form.set('operations', JSON.stringify(clone));
+
+      const map = {};
+      let i = 0;
+
+      files.forEach((paths) => {
+        map[i++] = paths;
+      });
+
+      form.set('map', JSON.stringify(map));
+
+      i = 0;
+      files.forEach((paths, file) => {
+        form.set(`${i++}`, file, file.name);
+      });
+
+      merge(init, {
+        body: form,
+      });
+    } else {
+      merge(init, {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(clone),
+      });
+    }
+
+    (async () => {
+      const request = new Request(httpEndpoint, init);
+
+      try {
+        const response = await fetch(request);
+
+        // Status code in range 200-299 inclusive (2xx).
+        if (response.ok) {
+          try {
+            const parts = await meros(response);
+
+            if (isAsyncIterable(parts)) {
+              for await (const part of parts) {
+                if (!part.json) {
+                  sink.error(
+                    new NetworkError(ErrorMessages.UNWORKABLE_FETCH, {
+                      request,
+                      response,
+                    }),
+                  );
+                  break;
+                }
+
+                // If any exceptions occurred when processing the request,
+                // throw an error to indicate to the developer what went wrong.
+                if (Array.isArray(part.body.errors)) {
+                  sink.error(
+                    new NetworkError(ErrorMessages.GRAPHQL_ERRORS, {
+                      request,
+                      response,
+                    }),
+                  );
+                  break;
+                }
+
+                // DEMO: delay chunked responses
+                // await pause(2_000);
+
+                // HACK: if `label` is present, also `path` is required, remove once fixed in HC
+                // @see https://github.com/maraisr/meros/issues/15#issuecomment-1054268416
+                if (part.body.label && !part.body.path) {
+                  part.body.path = [];
+                }
+
+                sink.next(part.body);
+              }
+            } else {
+              const json = await response.json();
+
+              // If any exceptions occurred when processing the request,
+              // throw an error to indicate to the developer what went wrong.
+              if (Array.isArray(json.errors)) {
+                sink.error(
+                  new NetworkError(ErrorMessages.GRAPHQL_ERRORS, {
+                    request,
+                    response,
+                  }),
+                );
+              } else {
+                // DEMO: delay response
+                // await pause(2_000);
+
+                sink.next(json);
+              }
+            }
+
+            sink.complete();
+          } catch (err) {
+            sink.error(
+              new NetworkError(ErrorMessages.UNWORKABLE_FETCH, {
+                cause: err,
+                request,
+                response,
+              }),
+              true,
+            );
+          }
+        } else {
+          sink.error(
+            new NetworkError(ErrorMessages.ERROR_FETCH, {request, response}),
+          );
+        }
+      } catch (err) {
+        sink.error(
+          new NetworkError(ErrorMessages.FAILED_FETCH, {cause: err, request}),
+          true,
+        );
+      }
+    })();
+  });
+};
+
+let wsClient;
+
+/**
+ * With `graphql-ws`.
+ * @see https://github.com/enisdenjo/graphql-ws
+ */
+const subscribeFn = (operation, variables) => {
+  const wsEndpoint = Config.WS_ENDPOINT;
+  const authToken = Config.AUTH_TOKEN;
+
+  const client = (wsClient ??= createClient({
+    url: wsEndpoint,
+    connectionParams: {
+      Authorization: authToken ? `basic ${authToken}` : undefined,
+    },
+  }));
+
+  return Observable.create((sink) =>
+    client.subscribe(
+      {
+        id: operation.id ?? undefined,
+        query: operation.text,
+        variables,
+      },
+      {
+        ...sink,
+        error: (err) => {
+          if (Array.isArray(err)) {
+            return sink.error(
+              new NetworkError(ErrorMessages.ERROR_FETCH, {cause: err}),
+            );
+          }
+
+          if (err instanceof CloseEvent) {
+            return sink.error(
+              new NetworkError(ErrorMessages.SOCKET_CLOSED, {cause: err}),
+            );
+          }
+
+          return sink.error(err, true);
+        },
+      },
+    ),
+  );
+};
+
+const createEnvironment = (initialRecords) => {
+  const source = new RecordSource(initialRecords);
+
+  /**
+   * Presence of Data
+   * @see https://relay.dev/docs/guided-tour/reusing-cached-data/presence-of-data/
+   *
+   * - Note that having a buffer size of 0 is equivalent to not having the release buffer, which means that queries will be immediately released and collected.
+   * - By default, environments have a release buffer size of 10.
+   *
+   * @example
+   * // last 10 queries
+   * gcReleaseBufferSize: 10,
+   *
+   *
+   * Staleness of Data
+   * @see https://relay.dev/docs/guided-tour/reusing-cached-data/staleness-of-data/
+   *
+   * - If the query cache expiration time is not provided, staleness checks only look at whether the referenced records have been invalidated.
+   *
+   * @example
+   * // 1 min
+   * queryCacheExpirationTime: 60 * 1_000,
+   */
+  const options = {};
+
+  const store = new Store(source, options);
+
+  const network = Network.create(fetchFn, subscribeFn);
+
+  return new Environment({
+    network,
+    store,
+  });
+};
+
+export const initEnvironment = (initialRecords) => {
+  // Create a network layer from the fetch function
+  const environment = relayEnvironment ?? createEnvironment(initialRecords);
+
+  // If your page has Next.js data fetching methods that use Relay, the initial records
+  // will get hydrated here
+  if (initialRecords) {
+    environment.getStore().publish(new RecordSource(initialRecords));
+  }
+  // For SSG and SSR always create a new Relay environment
+  if (typeof window === 'undefined') return environment;
+  // Create the Relay environment once in the client
+  if (!relayEnvironment) relayEnvironment = environment;
+
+  return relayEnvironment;
+};
+
+export const useEnvironment = (initialRecords) =>
+  useMemo(() => initEnvironment(initialRecords), [initialRecords]);
diff --git a/crypto/frontend/pages/_app.js b/crypto/frontend/pages/_app.js
index 8538cf3da3d9b697b1504a3947910a1823bea109..d430ee8773ac608e702e9e8de2c6123554c3745c 100644
--- a/crypto/frontend/pages/_app.js
+++ b/crypto/frontend/pages/_app.js
@@ -1,7 +1,9 @@
 import {CacheProvider} from '@emotion/react';
 import {CssBaseline, ThemeProvider} from '@mui/material';
 import {useEffect} from 'react';
+import {RelayEnvironmentProvider} from 'react-relay';
 
+import {useEnvironment} from '@/client';
 import {
   Content,
   Hacks,
@@ -19,6 +21,7 @@ export default function MyApp({
   emotionCache = clientSideEmotionCache,
   pageProps,
 }) {
+  const environment = useEnvironment(pageProps.initialRecords);
   const [theme, mode] = usePreferredTheme();
 
   useEffect(() => {
@@ -28,19 +31,21 @@ export default function MyApp({
   }, []);
 
   return (
-    <CacheProvider value={emotionCache}>
-      <ThemeProvider theme={theme}>
-        <ModeProvider value={mode}>
-          <Metadata />
-          <CssBaseline enableColorScheme />
-          <SlotsProvider>
-            <Content>
-              <Component {...pageProps} />
-            </Content>
-          </SlotsProvider>
-          <Hacks />
-        </ModeProvider>
-      </ThemeProvider>
-    </CacheProvider>
+    <RelayEnvironmentProvider environment={environment}>
+      <CacheProvider value={emotionCache}>
+        <ThemeProvider theme={theme}>
+          <ModeProvider value={mode}>
+            <Metadata />
+            <CssBaseline enableColorScheme />
+            <SlotsProvider>
+              <Content variant="crypto">
+                <Component {...pageProps} />
+              </Content>
+            </SlotsProvider>
+            <Hacks />
+          </ModeProvider>
+        </ThemeProvider>
+      </CacheProvider>
+    </RelayEnvironmentProvider>
   );
 }
diff --git a/crypto/frontend/scenes/dashboard/DashboardTicker.js b/crypto/frontend/scenes/dashboard/DashboardTicker.js
new file mode 100644
index 0000000000000000000000000000000000000000..fa89bb808ff2ac39bcbd5f9fd14e817b7546021f
--- /dev/null
+++ b/crypto/frontend/scenes/dashboard/DashboardTicker.js
@@ -0,0 +1,31 @@
+import {memo} from 'react';
+import {graphql, useFragment} from 'react-relay';
+
+import {Ticker} from '@/components';
+
+import DashboardTickerItem from './DashboardTickerItem';
+
+export default memo(function DashboardTicker({fragmentRef}) {
+  const data = useFragment(
+    graphql`
+      fragment DashboardTickerFragment_query on Query {
+        ticker: assets(first: 10) {
+          nodes {
+            symbol
+            ...DashboardTickerItemFragment_asset
+          }
+        }
+      }
+    `,
+    fragmentRef,
+  );
+  const assets = data.ticker?.nodes;
+
+  return (
+    <Ticker>
+      {assets?.map((asset) => (
+        <DashboardTickerItem key={asset.symbol} fragmentRef={asset} />
+      ))}
+    </Ticker>
+  );
+});
diff --git a/crypto/frontend/scenes/dashboard/DashboardContainer.js b/crypto/frontend/scenes/dashboard/DashboardContainer.js
new file mode 100644
index 0000000000000000000000000000000000000000..df4bb3b00bc1a4a5d5b71f71abe1adac35684f63
--- /dev/null
+++ b/crypto/frontend/scenes/dashboard/DashboardContainer.js
@@ -0,0 +1,28 @@
+import {Stack} from '@mui/material';
+import ErrorPage from 'next/error';
+import {memo} from 'react';
+import {graphql, useLazyLoadQuery} from 'react-relay';
+
+import DashboardTicker from './DashboardTicker';
+
+export default memo(function DashboardContainer({cacheBuster}) {
+  const data = useLazyLoadQuery(
+    graphql`
+      query DashboardContainerQuery {
+        ...DashboardTickerFragment_query
+      }
+    `,
+    {},
+    {fetchKey: cacheBuster},
+  );
+
+  if (!data) {
+    return <ErrorPage statusCode={404} title="Out of service" />;
+  }
+
+  return (
+    <Stack gap={2}>
+      <DashboardTicker fragmentRef={data} />
+    </Stack>
+  );
+});
diff --git a/crypto/frontend/schema/server.graphql b/crypto/frontend/schema/server.graphql
new file mode 100644
index 0000000000000000000000000000000000000000..3c6294443b568ab4461aa8f97ee47d981ed0a0c2
--- /dev/null
+++ b/crypto/frontend/schema/server.graphql
@@ -0,0 +1,309 @@
+"""
+The `@defer` directive may be provided for fragment spreads and inline fragments to inform the executor to delay the execution of the current fragment to indicate deprioritization of the current fragment. A query with `@defer` directive will cause the request to potentially return multiple responses, where non-deferred data is delivered in the initial response and data deferred is delivered in a subsequent response. `@include` and `@skip` take precedence over `@defer`.
+"""
+directive @defer(
+  """
+  Deferred when true.
+  """
+  if: Boolean
+
+  """
+  If this argument label has a value other than null, it will be passed on to the result of this defer directive. This label is intended to give client applications a way to identify to which fragment a deferred result belongs to.
+  """
+  label: String
+) on FRAGMENT_SPREAD | INLINE_FRAGMENT
+
+"""
+The `@stream` directive may be provided for a field of `List` type so that the backend can leverage technology such as asynchronous iterators to provide a partial list in the initial response, and additional list items in subsequent responses. `@include` and `@skip` take precedence over `@stream`.
+"""
+directive @stream(
+  """
+  Streamed when true.
+  """
+  if: Boolean
+
+  """
+  The initial elements that shall be send down to the consumer.
+  """
+  initialCount: Int! = 0
+
+  """
+  If this argument label has a value other than null, it will be passed on to the result of this stream directive. This label is intended to give client applications a way to identify to which fragment a streamed result belongs to.
+  """
+  label: String
+) on FIELD
+
+type Asset implements Node {
+  color: String!
+  description: String
+  id: ID!
+  imageUrl: String
+  name: String!
+  price: AssetPrice!
+  slug: String!
+  symbol: String!
+  website: String
+  whitePaper: String
+}
+
+input AssetFilterInput {
+  and: [AssetFilterInput!]
+  name: StringOperationFilterInput
+  or: [AssetFilterInput!]
+  price: AssetPriceFilterInput
+  slug: StringOperationFilterInput
+  symbol: StringOperationFilterInput
+}
+
+type AssetPrice implements Node {
+  asset: Asset!
+  change24Hour: Float!
+  circulatingSupply: Float!
+  currency: String!
+  high24Hour: Float!
+  id: ID!
+  lastPrice: Float!
+  low24Hour: Float!
+  marketCap: Float!
+  maxSupply: Float!
+  modifiedAt: DateTime
+  open24Hour: Float!
+  symbol: String!
+  tradableMarketCapRank: Float!
+  tradingActivity: Float!
+  volume24Hour: Float!
+  volumePercentChange24Hour: Float!
+}
+
+input AssetPriceFilterInput {
+  and: [AssetPriceFilterInput!]
+  asset: AssetFilterInput
+  assetId: ComparableInt32OperationFilterInput
+  change24Hour: ComparableDoubleOperationFilterInput
+  circulatingSupply: ComparableDoubleOperationFilterInput
+  currency: StringOperationFilterInput
+  high24Hour: ComparableDoubleOperationFilterInput
+  id: ComparableInt32OperationFilterInput
+  lastPrice: ComparableDoubleOperationFilterInput
+  low24Hour: ComparableDoubleOperationFilterInput
+  marketCap: ComparableDoubleOperationFilterInput
+  maxSupply: ComparableDoubleOperationFilterInput
+  modifiedAt: ComparableNullableOfDateTimeOperationFilterInput
+  open24Hour: ComparableDoubleOperationFilterInput
+  or: [AssetPriceFilterInput!]
+  symbol: StringOperationFilterInput
+  tradableMarketCapRank: ComparableDoubleOperationFilterInput
+  tradingActivity: ComparableDoubleOperationFilterInput
+  volume24Hour: ComparableDoubleOperationFilterInput
+  volumePercentChange24Hour: ComparableDoubleOperationFilterInput
+}
+
+input AssetPriceSortInput {
+  asset: AssetSortInput
+  assetId: SortEnumType
+  change24Hour: SortEnumType
+  circulatingSupply: SortEnumType
+  currency: SortEnumType
+  high24Hour: SortEnumType
+  id: SortEnumType
+  lastPrice: SortEnumType
+  low24Hour: SortEnumType
+  marketCap: SortEnumType
+  maxSupply: SortEnumType
+  modifiedAt: SortEnumType
+  open24Hour: SortEnumType
+  symbol: SortEnumType
+  tradableMarketCapRank: SortEnumType
+  tradingActivity: SortEnumType
+  volume24Hour: SortEnumType
+  volumePercentChange24Hour: SortEnumType
+}
+
+"""
+A connection to a list of items.
+"""
+type AssetsConnection {
+  """
+  A list of edges.
+  """
+  edges: [AssetsEdge!]
+
+  """
+  A flattened list of the nodes.
+  """
+  nodes: [Asset!]
+
+  """
+  Information to aid in pagination.
+  """
+  pageInfo: PageInfo!
+}
+
+"""
+An edge in a connection.
+"""
+type AssetsEdge {
+  """
+  A cursor for use in pagination.
+  """
+  cursor: String!
+
+  """
+  The item at the end of the edge.
+  """
+  node: Asset!
+}
+
+input AssetSortInput {
+  name: SortEnumType
+  price: AssetPriceSortInput
+  slug: SortEnumType
+  symbol: SortEnumType
+}
+
+input ComparableDoubleOperationFilterInput {
+  eq: Float
+  gt: Float
+  gte: Float
+  in: [Float!]
+  lt: Float
+  lte: Float
+  neq: Float
+  ngt: Float
+  ngte: Float
+  nin: [Float!]
+  nlt: Float
+  nlte: Float
+}
+
+input ComparableInt32OperationFilterInput {
+  eq: Int
+  gt: Int
+  gte: Int
+  in: [Int!]
+  lt: Int
+  lte: Int
+  neq: Int
+  ngt: Int
+  ngte: Int
+  nin: [Int!]
+  nlt: Int
+  nlte: Int
+}
+
+input ComparableNullableOfDateTimeOperationFilterInput {
+  eq: DateTime
+  gt: DateTime
+  gte: DateTime
+  in: [DateTime]
+  lt: DateTime
+  lte: DateTime
+  neq: DateTime
+  ngt: DateTime
+  ngte: DateTime
+  nin: [DateTime]
+  nlt: DateTime
+  nlte: DateTime
+}
+
+"""
+The `DateTime` scalar represents an ISO-8601 compliant date time type.
+"""
+scalar DateTime
+
+"""
+The node interface is implemented by entities that have a global unique identifier.
+"""
+interface Node {
+  id: ID!
+}
+
+"""
+Information about pagination in a connection.
+"""
+type PageInfo {
+  """
+  When paginating forwards, the cursor to continue.
+  """
+  endCursor: String
+
+  """
+  Indicates whether more edges exist following the set defined by the clients arguments.
+  """
+  hasNextPage: Boolean!
+
+  """
+  Indicates whether more edges exist prior the set defined by the clients arguments.
+  """
+  hasPreviousPage: Boolean!
+
+  """
+  When paginating backwards, the cursor to continue.
+  """
+  startCursor: String
+}
+
+type Query {
+  assetById(id: ID!): Asset
+  assets(
+    """
+    Returns the elements in the list that come after the specified cursor.
+    """
+    after: String
+
+    """
+    Returns the elements in the list that come before the specified cursor.
+    """
+    before: String
+
+    """
+    Returns the first _n_ elements from the list.
+    """
+    first: Int
+
+    """
+    Returns the last _n_ elements from the list.
+    """
+    last: Int
+  ): AssetsConnection
+
+  """
+  Fetches an object given its ID.
+  """
+  node(
+    """
+    ID of the object.
+    """
+    id: ID!
+  ): Node
+
+  """
+  Lookup nodes by a list of IDs.
+  """
+  nodes(
+    """
+    The list of node IDs.
+    """
+    ids: [ID!]!
+  ): [Node]!
+}
+
+enum SortEnumType {
+  ASC
+  DESC
+}
+
+input StringOperationFilterInput {
+  and: [StringOperationFilterInput!]
+  contains: String
+  endsWith: String
+  eq: String
+  in: [String]
+  ncontains: String
+  nendsWith: String
+  neq: String
+  nin: [String]
+  nstartsWith: String
+  or: [StringOperationFilterInput!]
+  startsWith: String
+}
diff --git a/crypto/frontend/scenes/settings/SettingsContainer.js b/crypto/frontend/scenes/settings/SettingsContainer.js
new file mode 100644
index 0000000000000000000000000000000000000000..bc2215e457518000d0af13f2a2a59c6b7a66fac7
--- /dev/null
+++ b/crypto/frontend/scenes/settings/SettingsContainer.js
@@ -0,0 +1,16 @@
+import {memo} from 'react';
+import {useState} from 'react';
+
+import SettingsCredentials from './SettingsCredentials';
+import SettingsLinks from './SettingsLinks';
+
+export default memo(function SettingsContainer() {
+  const [active, setActive] = useState('profile');
+
+  return (
+    <>
+      <SettingsLinks active={active} onChange={setActive} />
+      <SettingsCredentials active={active} onChange={setActive} />
+    </>
+  );
+});
diff --git a/crypto/frontend/pages/dashboard.js b/crypto/frontend/pages/dashboard.js
new file mode 100644
index 0000000000000000000000000000000000000000..f5b3bfcc5a809b3d23c30b8a2d16fc34f5eabf4a
--- /dev/null
+++ b/crypto/frontend/pages/dashboard.js
@@ -0,0 +1 @@
+export {Dashboard as default} from '@/scenes';
